### PR TITLE
Add match and match phrase queries

### LINKED ISSUE #47: Add match and match phrase queries
Match and MatchPhrase are Elasticsearch query types which analyze the query text to create a boolean query consisting of terms or phrase queries. We can create equivalent queries using `MultiFieldQueryParser` in Lucene.

### PR DESCRIPTION
**Pull request by sarthakn7**
Fixes #47 .

### CODE REVIEW

File: src/test/java/org/apache/platypus/server/grpc/LuceneServerTest.java [hunk 1]
```diff
@@ -466,6 +490,66 @@ public void testSearchDisjunctionMaxQuery() throws IOException, InterruptedExcep
...
+        //manual refresh
+        grpcServer.getBlockingStub().refresh(RefreshRequest.newBuilder().setIndexName(grpcServer.getTestIndex()).build());
+
+        SearchResponse searchResponse = grpcServer.getBlockingStub()
+                .search(SearchRequest.newBuilder()
+                        .setIndexName(grpcServer.getTestIndex())
+                        .setStartHit(0)
+                        .setTopHits(10)
+                        .addAllRetrieveFields(RETRIEVED_VALUES)
+                        .setQuery(Query.newBuilder()
```
Comment (umeshdangat) [inline src/test/java/org/apache/platypus/server/grpc/LuceneServerTest.java]: Maybe add tests for custom user input analyzer? Same for MatchPhraseQuery

File: src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java [hunk 2]
```diff
@@ -106,6 +115,40 @@ private DisjunctionMaxQuery getDisjunctionMaxQuery(org.apache.platypus.server.gr
...
 
+    private Query getMatchQuery(MatchQuery matchQuery, IndexState state) {
+        Analyzer analyzer = isAnalyzerDefined(matchQuery.getAnalyzer())
+                ? getAnalyzer(matchQuery.getAnalyzer())
+                : state.searchAnalyzer;
+
+        MultiFieldQueryParser multiFieldQueryParser = new MultiFieldQueryParser(null, analyzer);
+
+        // This created query will be TermQuery if only one token is found after analysis, otherwise BooleanQuery
+        Query query = multiFieldQueryParser.createBooleanQuery(matchQuery.getField(), matchQuery.getQuery(), occurMapping.get(matchQuery.getOperator()));
```
Comment (umeshdangat) [inline src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java]: Seems like the base class QueryBuilder throws exception for any clauses other than MUST and SHOULD. What do you think about making a more restrictive proto clause with only those two options. Might be easier on the user to not worry about RuntimeError due to invalid clause.

```
    if (operator != BooleanClause.Occur.SHOULD && operator != BooleanClause.Occur.MUST) {
      throw new IllegalArgumentException("invalid operator: only SHOULD or MUST are allowed");
    }
```

File: src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java [hunk 3]
```diff
@@ -106,6 +115,40 @@ private DisjunctionMaxQuery getDisjunctionMaxQuery(org.apache.platypus.server.gr
         return new DisjunctionMaxQuery(disjuncts, disjunctionMaxQuery.getTieBreakerMultiplier());
     }
 
+    private Query getMatchQuery(MatchQuery matchQuery, IndexState state) {
+        Analyzer analyzer = isAnalyzerDefined(matchQuery.getAnalyzer())
+                ? getAnalyzer(matchQuery.getAnalyzer())
+                : state.searchAnalyzer;
+
+        MultiFieldQueryParser multiFieldQueryParser = new MultiFieldQueryParser(null, analyzer);
```
Comment (umeshdangat) [inline src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java]: Would the input queries not have boosts? Seems like MultiFieldQueryParser supports boosts on fields as well.

File: src/test/java/org/apache/platypus/server/grpc/LuceneServerTest.java (hunk 1 above)
Comment (sarthakn7) [reply src/test/java/org/apache/platypus/server/grpc/LuceneServerTest.java]: Added

File: src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java (hunk 2 above)
Comment (sarthakn7) [reply src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java]: Added MatchOperator with only MUST and SHOULD.

File: src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java (hunk 3 above)
Comment (sarthakn7) [reply src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java]: I will add a query level boost separately (and per-field boost for multi-match).

**Pull request merged**